drivers/sensor/vl53l1x/vl53l1.c

```diff
@@ -297,35 +297,81 @@ static int vl53l1x_get_roi(const struct device *dev,
 	return 0;
 }
 
+static int vl53l1x_get_inter_measurement_freq(const struct device *dev, struct sensor_value *val)
+{
+	struct vl53l1x_data *drv_data = dev->data;
+
+	uint32_t period_ms;
+	VL53L1_Error ret = VL53L1_GetInterMeasurementPeriodMilliSeconds(&drv_data->vl53l1x, &period_ms);
+
+	if (ret != VL53L1_ERROR_NONE) {
+		LOG_ERR("[%s] VL53L1_GetInterMeasurementPeriodMilliSeconds return error (%d)", dev->name, ret);
+		return -ENODATA;
+	}
+
+	/* Map period_ms to val in Hz */
+	val->val1 = (int32_t) (1000 / period_ms);
+	val->val2 = 0;
+
+	return 0;
+}
+
 static int vl53l1x_sample_fetch(const struct device *dev,
 		enum sensor_channel chan)
 {
 	struct vl53l1x_data *drv_data = dev->data;
 	VL53L1_Error ret;
 
-	__ASSERT_NO_MSG(chan == SENSOR_CHAN_DISTANCE);
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
+			chan == SENSOR_CHAN_DISTANCE ||
+			chan == SENSOR_CHAN_PROX);
 
-	/* Will immediately stop current measurement */
-	ret = VL53L1_StopMeasurement(&drv_data->vl53l1x);
-	if (ret != VL53L1_ERROR_NONE) {
-		LOG_ERR("VL53L1_StopMeasurement return error (%d)", ret);
-		return -EBUSY;
-	}
+	/* if ranging mode, we need to start a new measurement */
+	if (drv_data->autonomous_intermeasurement_period_ms == 0) {
+		/* Will immediately stop current measurement */
+		ret = VL53L1_StopMeasurement(&drv_data->vl53l1x);
+		if (ret != VL53L1_ERROR_NONE) {
+			LOG_ERR("VL53L1_StopMeasurement return error (%d)", ret);
+			return -EBUSY;
+		}
 
 #ifdef CONFIG_VL53L1X_INTERRUPT_MODE
-	const struct vl53l1x_config *config = dev->config;
+		const struct vl53l1x_config *config = dev->config;
 
-	ret = gpio_pin_interrupt_configure_dt(&config->gpio1, GPIO_INT_EDGE_TO_INACTIVE);
-	if (ret < 0) {
-		LOG_ERR("[%s] Unable to config interrupt", dev->name);
-		return -EIO;
-	}
+		ret = gpio_pin_interrupt_configure_dt(&config->gpio1, GPIO_INT_EDGE_TO_INACTIVE);
+		if (ret < 0) {
+			LOG_ERR("[%s] Unable to config interrupt", dev->name);
+			return -EIO;
+		}
 #endif
 
-	ret = VL53L1_StartMeasurement(&drv_data->vl53l1x);
-	if (ret != VL53L1_ERROR_NONE) {
-		LOG_ERR("[%s] VL53L1_StartMeasurement return error (%d)", dev->name, ret);
-		return -EBUSY;
+		ret = VL53L1_StartMeasurement(&drv_data->vl53l1x);
+		if (ret != VL53L1_ERROR_NONE) {
+			LOG_ERR("[%s] VL53L1_StartMeasurement return error (%d)", dev->name, ret);
+			return -EBUSY;
+		}
+	} else {
+		if (IS_ENABLED(CONFIG_VL53L1X_INTERRUPT_MODE) == 0) {
```

Comment:
```
 * That is, it works similarly to <tt>\#if defined(CONFIG_FOO)</tt>
 * except that its expansion is a C expression. Thus, much <tt>\#ifdef</tt>
 * usage can be replaced with equivalents like:
 *
 *     if (IS_ENABLED(CONFIG_FOO)) {
 *             do_something_with_foo
 *     }
 *
 * This is cleaner since the compiler can generate errors and warnings
 * for @p do_something_with_foo even when @p CONFIG_FOO is undefined.
```